### PR TITLE
Add frustum query in octree and frustum query client

### PR DESCRIPTION
**Pull request by ksavinash9**
Augment octree server to to:
* Add Frustum Query GRPC API
* Add frustum query client for verification

### CODE REVIEW

**Comment by ksavinash9** [comment]
@SirVer Please review.

File: point_viewer_grpc/src/bin/octree_server.rs [hunk 1]
```diff
@@ -30,6 +30,108 @@ struct OctreeService {
...
+    query: OctreeQuery,
+    octree: Arc<OnDiskOctree>,
+    ctx: RpcContext,
+    resp: ServerStreamingSink<proto::PointsReply>)
+{
+    use thread;
+
+    // This creates a async-aware (tx, rx) pair that can wake up the event loop when new data
+    // is piped through it.
+    let (tx, rx) = mpsc::channel(4);
```
Comment (SirVer) [inline point_viewer_grpc/src/bin/octree_server.rs]: Add a comment for the 4: E.g.

> We create a channel with a small buffer, which yields better performance than fully blocking without requiring a ton of memory. This has not been carefully benchmarked for best performance though.

Comment (ksavinash9) [reply point_viewer_grpc/src/bin/octree_server.rs]: Added

**Comment by SirVer** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user SirVer. Merge queue now has a length of 1.

**Comment by wally-the-cartographer** [comment]
Error: write /dev/stdout: input/output error

Log:
~~~
=> handle work item: PR 156 on googlecartographer/point_cloud_viewer.
 => Switching to branch master.
 $ git reset --hard HEAD [data/point_cloud_viewer]
HEAD is now at 995a146 Refactor octree server grpc response proto (#158)

~~~

**Comment by ksavinash9** [comment]
@cschuet Wally isn't working. Please, merge.

**Pull request merged**